freedomofpress/securedrop-debian-packaging
# Makefile updates and tools for requirements.txt and wheels

This PR has a few new scripts and related Makefile update. @conorsch I am guessing you will update the Makefile bit to make a bit more professional :)

These scripts and Makefile targets will help our package maintainers to create `requirements.txt` and `requirements-dev.txt` files for the actual release. Those files will contain the sha256sums for the binary wheels from our PyPI (the s3 bucket).

### How to test?


First, let us the create/update the files, requires `PKG_DIR` environment variable.

```
PKG_DIR=/home/user/code/securedrop-client make requirements
```

This command will update the requirements/versions from the upstream PyPI. If there is any new update or if any of the new binary wheels are missing, this will fail. It will also print the next command a developer can use to generate the missing wheels.

```
$ PKG_DIR=/home/user/code/securedrop-client ./scripts/update-requirements 
The following dependent wheel(s) are missing:
pytest==3.10.1

Please build the wheel by using the following command:

	PKG_DIR=/home/user/code/securedrop-client make build-wheels

Then sync the newly built wheels and sources to the s3 bucket.
Also update the index HTML files accordingly and sync to s3.
After these steps, please rerun the command again.
```
If such missing wheels are there (you can delete any related wheel and test this), the above mentioned command will create such new wheels.

```
PKG_DIR=/home/user/code/securedrop-client make build-wheels
```

After this, remember to sync the `localwheels` directory to the s3, and also update the HTML index files and also the `sha256sums.txt` files as required. Also remember to deploy any new wheel+index to the s3 bucket.

The `sha256sums.txt` should always get new entries, but, there should not be any update of the any existing wheels or hashes. If there is any such change, it means one did a rebuild of the wheel which is already there in our s3/PyPI index. So, from the security aspect, we will have to check that in the CI for each commit.
